app/bundles/PageBundle/Event/UrlTokenReplaceEvent.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\PageBundle\Event;
+
+use Mautic\LeadBundle\Entity\Lead;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * Event dispatched when replacing tokens in a URL, allowing modifications before the final redirect.
+ */
+class UrlTokenReplaceEvent extends Event
+{
+    public function __construct(
+        private string $content,
+        private Lead|int|null $lead,
```

Comment:
I believe the `$lead` can be only type of `Lead`. I don't see any other option.
```suggestion
        private Lead $lead,
```
Would it even make sense to trigger this event with no Lead?